kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchConnectionManager.java

```diff
@@ -120,9 +120,9 @@ public void onOpen(WebSocket webSocket, Response response) {
       @Override
       public void onFailure(IOException e, Response response) {
         if (!forceClosed.get()) {
-          if (response != null) {
-            try {
-              Status responseStatus = mapper.readValue(response.body().byteStream(), Status.class);
+          if (response != null && response.body() != null) {
+            try (ResponseBody body = response.body()) {
```

Comment:
Is this definitely required? I thought Jackson closed the underlying stream with readValue so closing the body directly would be a no-op? If it is required then we need to do this everywhere we unmarshal responses.